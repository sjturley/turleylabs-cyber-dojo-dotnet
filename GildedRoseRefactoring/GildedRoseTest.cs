using NUnit.Framework;
using System;

[TestFixture]
public class GildedRoseTest
{
    [Test]
    public void ApproveGildedRose()
    {
    }
}