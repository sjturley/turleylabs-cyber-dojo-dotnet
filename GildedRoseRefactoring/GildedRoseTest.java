import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.reporters.QuietReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

@UseReporter(QuietReporter.class)
class GildedRoseTest {

    @Test
    void approveGildedRose() {
        String[] names = {"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"};
        Integer[] sellins = { -1, 0, 12, 6, 2 };
        Integer[] qualities = { 0, 1, 48, 49, 50 };
        CombinationApprovals.verifyAllCombinations(
                this::runGildedRose,
                names,
                sellins,
                qualities);
    }

    private Item runGildedRose(String name, int sellIn, int quality) {
        Item[] items = new Item[] { new Item(name, sellIn, quality) };
        GildedRose gildedRose = new GildedRose(items);
        gildedRose.updateItems();
        return gildedRose.items[0];
    }

}
