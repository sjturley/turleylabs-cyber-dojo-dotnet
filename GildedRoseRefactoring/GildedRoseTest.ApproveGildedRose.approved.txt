[foo,-1,0] => Item{name='foo', sellIn=-2, quality=0}
[foo,-1,1] => Item{name='foo', sellIn=-2, quality=0}
[foo,-1,48] => Item{name='foo', sellIn=-2, quality=46}
[foo,-1,49] => Item{name='foo', sellIn=-2, quality=47}
[foo,-1,50] => Item{name='foo', sellIn=-2, quality=48}
[foo,0,0] => Item{name='foo', sellIn=-1, quality=0}
[foo,0,1] => Item{name='foo', sellIn=-1, quality=0}
[foo,0,48] => Item{name='foo', sellIn=-1, quality=46}
[foo,0,49] => Item{name='foo', sellIn=-1, quality=47}
[foo,0,50] => Item{name='foo', sellIn=-1, quality=48}
[foo,12,0] => Item{name='foo', sellIn=11, quality=0}
[foo,12,1] => Item{name='foo', sellIn=11, quality=0}
[foo,12,48] => Item{name='foo', sellIn=11, quality=47}
[foo,12,49] => Item{name='foo', sellIn=11, quality=48}
[foo,12,50] => Item{name='foo', sellIn=11, quality=49}
[foo,6,0] => Item{name='foo', sellIn=5, quality=0}
[foo,6,1] => Item{name='foo', sellIn=5, quality=0}
[foo,6,48] => Item{name='foo', sellIn=5, quality=47}
[foo,6,49] => Item{name='foo', sellIn=5, quality=48}
[foo,6,50] => Item{name='foo', sellIn=5, quality=49}
[foo,2,0] => Item{name='foo', sellIn=1, quality=0}
[foo,2,1] => Item{name='foo', sellIn=1, quality=0}
[foo,2,48] => Item{name='foo', sellIn=1, quality=47}
[foo,2,49] => Item{name='foo', sellIn=1, quality=48}
[foo,2,50] => Item{name='foo', sellIn=1, quality=49}
[Aged Brie,-1,0] => Item{name='Aged Brie', sellIn=-2, quality=4}
[Aged Brie,-1,1] => Item{name='Aged Brie', sellIn=-2, quality=5}
[Aged Brie,-1,48] => Item{name='Aged Brie', sellIn=-2, quality=50}
[Aged Brie,-1,49] => Item{name='Aged Brie', sellIn=-2, quality=50}
[Aged Brie,-1,50] => Item{name='Aged Brie', sellIn=-2, quality=50}
[Aged Brie,0,0] => Item{name='Aged Brie', sellIn=-1, quality=4}
[Aged Brie,0,1] => Item{name='Aged Brie', sellIn=-1, quality=5}
[Aged Brie,0,48] => Item{name='Aged Brie', sellIn=-1, quality=50}
[Aged Brie,0,49] => Item{name='Aged Brie', sellIn=-1, quality=50}
[Aged Brie,0,50] => Item{name='Aged Brie', sellIn=-1, quality=50}
[Aged Brie,12,0] => Item{name='Aged Brie', sellIn=11, quality=1}
[Aged Brie,12,1] => Item{name='Aged Brie', sellIn=11, quality=2}
[Aged Brie,12,48] => Item{name='Aged Brie', sellIn=11, quality=49}
[Aged Brie,12,49] => Item{name='Aged Brie', sellIn=11, quality=50}
[Aged Brie,12,50] => Item{name='Aged Brie', sellIn=11, quality=50}
[Aged Brie,6,0] => Item{name='Aged Brie', sellIn=5, quality=2}
[Aged Brie,6,1] => Item{name='Aged Brie', sellIn=5, quality=3}
[Aged Brie,6,48] => Item{name='Aged Brie', sellIn=5, quality=50}
[Aged Brie,6,49] => Item{name='Aged Brie', sellIn=5, quality=50}
[Aged Brie,6,50] => Item{name='Aged Brie', sellIn=5, quality=50}
[Aged Brie,2,0] => Item{name='Aged Brie', sellIn=1, quality=3}
[Aged Brie,2,1] => Item{name='Aged Brie', sellIn=1, quality=4}
[Aged Brie,2,48] => Item{name='Aged Brie', sellIn=1, quality=50}
[Aged Brie,2,49] => Item{name='Aged Brie', sellIn=1, quality=50}
[Aged Brie,2,50] => Item{name='Aged Brie', sellIn=1, quality=50}
[Backstage passes to a TAFKAL80ETC concert,-1,0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0}
[Backstage passes to a TAFKAL80ETC concert,-1,1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0}
[Backstage passes to a TAFKAL80ETC concert,-1,48] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0}
[Backstage passes to a TAFKAL80ETC concert,-1,49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0}
[Backstage passes to a TAFKAL80ETC concert,-1,50] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0}
[Backstage passes to a TAFKAL80ETC concert,0,0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0}
[Backstage passes to a TAFKAL80ETC concert,0,1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0}
[Backstage passes to a TAFKAL80ETC concert,0,48] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0}
[Backstage passes to a TAFKAL80ETC concert,0,49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0}
[Backstage passes to a TAFKAL80ETC concert,0,50] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0}
[Backstage passes to a TAFKAL80ETC concert,12,0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=11, quality=1}
[Backstage passes to a TAFKAL80ETC concert,12,1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=11, quality=2}
[Backstage passes to a TAFKAL80ETC concert,12,48] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=11, quality=49}
[Backstage passes to a TAFKAL80ETC concert,12,49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=11, quality=50}
[Backstage passes to a TAFKAL80ETC concert,12,50] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=11, quality=50}
[Backstage passes to a TAFKAL80ETC concert,6,0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=5, quality=1}
[Backstage passes to a TAFKAL80ETC concert,6,1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=5, quality=2}
[Backstage passes to a TAFKAL80ETC concert,6,48] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=5, quality=49}
[Backstage passes to a TAFKAL80ETC concert,6,49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=5, quality=50}
[Backstage passes to a TAFKAL80ETC concert,6,50] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=5, quality=50}
[Backstage passes to a TAFKAL80ETC concert,2,0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=1, quality=1}
[Backstage passes to a TAFKAL80ETC concert,2,1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=1, quality=2}
[Backstage passes to a TAFKAL80ETC concert,2,48] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=1, quality=49}
[Backstage passes to a TAFKAL80ETC concert,2,49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=1, quality=50}
[Backstage passes to a TAFKAL80ETC concert,2,50] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=1, quality=50}
[Sulfuras, Hand of Ragnaros,-1,0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=0}
[Sulfuras, Hand of Ragnaros,-1,1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=1}
[Sulfuras, Hand of Ragnaros,-1,48] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=48}
[Sulfuras, Hand of Ragnaros,-1,49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=49}
[Sulfuras, Hand of Ragnaros,-1,50] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=50}
[Sulfuras, Hand of Ragnaros,0,0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=0}
[Sulfuras, Hand of Ragnaros,0,1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=1}
[Sulfuras, Hand of Ragnaros,0,48] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=48}
[Sulfuras, Hand of Ragnaros,0,49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=49}
[Sulfuras, Hand of Ragnaros,0,50] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=50}
[Sulfuras, Hand of Ragnaros,12,0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=12, quality=0}
[Sulfuras, Hand of Ragnaros,12,1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=12, quality=1}
[Sulfuras, Hand of Ragnaros,12,48] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=12, quality=48}
[Sulfuras, Hand of Ragnaros,12,49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=12, quality=49}
[Sulfuras, Hand of Ragnaros,12,50] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=12, quality=50}
[Sulfuras, Hand of Ragnaros,6,0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=6, quality=0}
[Sulfuras, Hand of Ragnaros,6,1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=6, quality=1}
[Sulfuras, Hand of Ragnaros,6,48] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=6, quality=48}
[Sulfuras, Hand of Ragnaros,6,49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=6, quality=49}
[Sulfuras, Hand of Ragnaros,6,50] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=6, quality=50}
[Sulfuras, Hand of Ragnaros,2,0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=2, quality=0}
[Sulfuras, Hand of Ragnaros,2,1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=2, quality=1}
[Sulfuras, Hand of Ragnaros,2,48] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=2, quality=48}
[Sulfuras, Hand of Ragnaros,2,49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=2, quality=49}
[Sulfuras, Hand of Ragnaros,2,50] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=2, quality=50}
